pkg/destroy/openstack/openstack.go

```diff
@@ -998,3 +1009,66 @@ func deleteImages(opts *clientconfig.ClientOpts, filter Filter, logger logrus.Fi
 	}
 	return true, nil
 }
+
+func untagRunner(opts *clientconfig.ClientOpts, networkTag string, logger logrus.FieldLogger) error {
+	backoffSettings := wait.Backoff{
+		Duration: time.Second * 10,
+		Steps:    25,
+	}
+
+	err := wait.ExponentialBackoff(backoffSettings, func() (bool, error) {
+		return untagNetwork(opts, networkTag, logger)
+	})
+	if err != nil {
+		if err == wait.ErrWaitTimeout {
+			return err
+		}
+		return errors.Errorf("Unrecoverable error: %v", err)
+	}
+
+	return nil
+}
+
+// untagNetwork removes a tag from the network
+func untagNetwork(opts *clientconfig.ClientOpts, networkTag string, logger logrus.FieldLogger) (bool, error) {
+	logger.Debugf("Removing tag %v from openstack networks", networkTag)
+	defer logger.Debug("Exiting untagging openstack networks")
+
+	conn, err := clientconfig.NewServiceClient("network", opts)
+	if err != nil {
+		logger.Debug(err)
+		return false, nil
+	}
+
+	listOpts := networks.ListOpts{
+		Tags: networkTag,
+	}
+
+	allPages, err := networks.List(conn, listOpts).AllPages()
+	if err != nil {
+		logger.Debug(err)
+		return false, nil
+	}
+
+	allNetworks, err := networks.ExtractNetworks(allPages)
+	if err != nil {
+		logger.Debug(err)
+		return false, nil
+	}
+
+	if len(allNetworks) > 1 {
+		return false, errors.Errorf("More than one network with tag %v", networkTag)
```

Comment:
Why make it an error? That will cause the untagRunner to enter a loop to perform an action it can't do.
IMO, as opposed to deleting resources when we're not sure we were the ones creating them, it's OK to remove (internally used) tags from resources.

Reply:
If we return an error, then untagRunner stops working immediately, no matter how many attempts there are left. So, for the runner it basically means - stop looping and return the error.
https://github.com/kubernetes/apimachinery/blob/release-1.17/pkg/util/wait/wait.go#L293

Despite the fact we consider the tag to be unique because it contains an infra ID, users can still add the tag to other networks. So removing them from all networks can lead to unpleasant side effects, which means we must stop the execution and do not try to repeat the untagging.